components/demo/demo-snippet.js

```diff
@@ -30,12 +66,14 @@ class DemoSnippet extends LitElement {
 	render() {
 		const dirAttr = this._dir === 'rtl' ? 'rtl' : 'ltr';
 		return html`
-			<div class="d2l-demo-snippet-demo" dir="${dirAttr}">
-				<div class="d2l-demo-snippet-actions">
-					<button id="d2l-demo-snippet-toggle-dir" @click="${this._handleDirChange}" title="toggle dir">${this._dirButton}</button>
+			<div class="d2l-demo-snippet-demo-wrapper">
+				<div class="d2l-demo-snippet-demo" dir="${dirAttr}">
+					<slot name="_demo"></slot>
+					<slot></slot>
+				</div>
+				<div class="d2l-demo-snippet-settings">
```

Comment:
Not that it need to be handled in this PR, but another use-case that comes up in some of our demo pages are ad hoc actions.  Many times we end up adding extra buttons outside the snippet for testing purposes.  ex.

* a "focus" button that applies focus to the component (html editor)
* buttons for adding/removing content (floating buttons, tabs demos)
* buttons for manipulating hierarchical views

Many times these buttons, and their associated JS could be placed in the snippet, but the choice was made not to in order to keep the snippet code focused.

So, I am wonder if we would find it useful to have a slot to help group these things.  Focus is one that ought to be more common than it actually is, and perhaps we should build it in.  Sometimes though, the snippet contains several elements, like a menu + menu-items, and the first element isn't necessarily the one we're interested in applying settings to.

Reply:
Interesting, yeah this could easily have a slot and/or come with some built-in actions. I'll think about this more and it'll come up again when we look to improve this further for the component catalog.